packages/aws-cdk/lib/api/cloudformation-deployments.ts

```diff
@@ -298,21 +296,15 @@ export class CloudFormationDeployments {
 
   public async readCurrentTemplateWithNestedStacks(rootStackArtifact: cxapi.CloudFormationStackArtifact): Promise<Template> {
     const sdk = await this.prepareSdkWithLookupOrDeployRole(rootStackArtifact);
-    const deployedTemplate = await this.readCurrentTemplate(rootStackArtifact, sdk);
-    await this.addNestedTemplatesToGeneratedAndDeployedStacks(rootStackArtifact, sdk, {
-      generatedTemplate: rootStackArtifact.template,
-      deployedTemplate: deployedTemplate,
-      deployedStackName: rootStackArtifact.stackName,
-    });
-    return deployedTemplate;
+    return (await loadCurrentTemplateWithNestedStacks(rootStackArtifact, sdk)).deployedTemplate;
   }
 
   public async readCurrentTemplate(stackArtifact: cxapi.CloudFormationStackArtifact, sdk?: ISDK): Promise<Template> {
```

Comment:
Am I wrong in thinking you can get rid of the `sdk` parameter now? This method's [only remaining caller](https://github.com/aws/aws-cdk/blob/19ead77c6470c0aca4900c4ee200a7c4e66ebbed/packages/aws-cdk/lib/cdk-toolkit.ts#L183-L183) doesn't use it (it used to be called from `readCurrentTemplateWithNestedStacks()`, but that's not the case anymore).